### PR TITLE
116842 initial checking system

### PR DESCRIPTION
**Pull request by jesseeproductions**
_Ref:_ [C#116841](https://central.tri.be/issues/116841)

Dependency Checker PR

Please follow the order below to review as I tried to keep the coding the same throughout.

Some notes to help explain what the changes entail:

Dev WriteUp
https://docs.google.com/document/d/1mk1rNpdRnZwAMrGQs1wY60oUhg1Zvl1WVVwWAj4Eg9U/edit#heading=h.h9mgoxmdqvd3

For ET and TEC if Dependency checker version of those plugins loads and the other plugin is before this release it will shut itself down and display a message to update the other plugin. The older plugin will fully load. This happens to prevent fatal errors as Dependency checker changes the timing of the loading and when it tries to load with an older version it crashes. 

Except for the above, ET and TEC will almost always load and only add-ons will stop loading.

Add-ons can be prevented from loading from themselves, other add-ons, and parents (et and tec)

I deprecated the min common version as checking for a minimum version of ET or TEC will do the same thing and common will load the highest version available once all are on at least dependency checker versions. Checking for a minimum ET or TEC version is a check for a minimum common version as they are always coupled together. 

The dependency admin message can display all the plugins that need to be updated, for instance if CT needs all 4 of its dependencies updated, it will be included them all in one the admin message.

The admin message also includes the minimum version a plugin needs to be updated too next to the plugin name. 

QA Grid
https://docs.google.com/spreadsheets/u/2/d/1JaplcCToJEKE5SiSOU4fcKg_4Ro1C0zviDDClUgrbSQ/edit?ouid=115512406234995825607&usp=sheets_home&ths=true

Please review in this order:

Common and TEC:
https://github.com/moderntribe/tribe-common/pull/900
https://github.com/moderntribe/the-events-calendar/pull/2353

PRO
https://github.com/moderntribe/events-pro/pull/1096

Common and ET
https://github.com/moderntribe/event-tickets/pull/1038

ET+
https://github.com/moderntribe/event-tickets-plus/pull/665

The above should all be merged before reviewing:

Filter
https://github.com/moderntribe/events-filterbar/pull/159/files

EB
https://github.com/moderntribe/events-eventbrite/pull/273

CE
https://github.com/moderntribe/events-community/pull/353

CT
https://github.com/moderntribe/events-community-tickets/pull/122

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1429](https://github.com/moderntribe/the-events-calendar/blob/dd4af2f0fde13da6e25e051d350c39a7f02928f2/src/Tribe/Main.php#L1429)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1432](https://github.com/moderntribe/the-events-calendar/blob/dd4af2f0fde13da6e25e051d350c39a7f02928f2/src/Tribe/Main.php#L1432)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1429](https://github.com/moderntribe/the-events-calendar/blob/a87daf97747368972ece59e7ee39258ea73a240f/src/Tribe/Main.php#L1429)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1432](https://github.com/moderntribe/the-events-calendar/blob/a87daf97747368972ece59e7ee39258ea73a240f/src/Tribe/Main.php#L1432)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1431](https://github.com/moderntribe/the-events-calendar/blob/df58056642c911a46d08e30412019170429fe064/src/Tribe/Main.php#L1431)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1434](https://github.com/moderntribe/the-events-calendar/blob/df58056642c911a46d08e30412019170429fe064/src/Tribe/Main.php#L1434)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -4874,11 +4802,69 @@ public function remove_preview_organizers( $event_id, $delete_meta = false ) {
...
+		 *
+		 * @since  4.7
+		 *
+		 * @return boolean [description]
+		 */
+		public function is_delayed_activation() {
+/*			_deprecated_function(
+				__METHOD__,
+				'',
+				'TBD'
```
Comment (ggwicz) [inline src/Tribe/Main.php]: This _deprecated_function() code block is commented out—is that intentional? More work needed here? If not, remove it

File: src/Tribe/Main.php [hunk 2]
```diff
@@ -4874,11 +4802,69 @@ public function remove_preview_organizers( $event_id, $delete_meta = false ) {
...
+		 * @deprecated TBD
+		 *
+		 * @since  4.7
+		 *
+		 * @return bool
+		 */
+		public function maybe_delay_activation_if_outdated_common() {
+/*			_deprecated_function(
+				__METHOD__,
+				'',
```
Comment (ggwicz) [inline src/Tribe/Main.php]: ditto with this _deprecated_function block as the one above

File: src/Tribe/Main.php [hunk 3]
```diff
@@ -329,80 +357,42 @@ public function maybe_set_common_lib_info() {
...
-			if ( empty( $GLOBALS['tribe-common-info'] ) ) {
-				return false;
+				return;
 			}
 
-			// Only when this common lib is newer than the loaded one on activation we bail
-			if ( ! version_compare( $GLOBALS['tribe-common-info']['version'], $common_version, '>' ) ) {
-				return false;
+			// WordPress and PHP Version Check
+			if ( ! self::supportedVersion( 'wordpress' ) || ! self::supportedVersion( 'php' ) ) {
```
[the remark this answers is not in the document]
Comment (jesseeproductions) [reply src/Tribe/Main.php]: For sure, but this is an existing method.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1431](https://github.com/moderntribe/the-events-calendar/blob/5089a2784d0b81622e0c5e02a591d7b3de823c53/src/Tribe/Main.php#L1431)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1434](https://github.com/moderntribe/the-events-calendar/blob/5089a2784d0b81622e0c5e02a591d7b3de823c53/src/Tribe/Main.php#L1434)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1431](https://github.com/moderntribe/the-events-calendar/blob/5089a2784d0b81622e0c5e02a591d7b3de823c53/src/Tribe/Main.php#L1431)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1434](https://github.com/moderntribe/the-events-calendar/blob/5089a2784d0b81622e0c5e02a591d7b3de823c53/src/Tribe/Main.php#L1434)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1431](https://github.com/moderntribe/the-events-calendar/blob/c1cf4cfde01dc28257ad5ae1501168f4056bb5ca/src/Tribe/Main.php#L1431)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1434](https://github.com/moderntribe/the-events-calendar/blob/c1cf4cfde01dc28257ad5ae1501168f4056bb5ca/src/Tribe/Main.php#L1434)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1431](https://github.com/moderntribe/the-events-calendar/blob/d68e0783fb6deff40e74008d01b1ac26924c2f26/src/Tribe/Main.php#L1431)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1434](https://github.com/moderntribe/the-events-calendar/blob/d68e0783fb6deff40e74008d01b1ac26924c2f26/src/Tribe/Main.php#L1434)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1432](https://github.com/moderntribe/the-events-calendar/blob/b269daafc89d3a6e298fdbdc84f5c6ab71a39160/src/Tribe/Main.php#L1432)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1435](https://github.com/moderntribe/the-events-calendar/blob/b269daafc89d3a6e298fdbdc84f5c6ab71a39160/src/Tribe/Main.php#L1435)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1432](https://github.com/moderntribe/the-events-calendar/blob/9835f93e6c067ea78338cf99f5aa535c970c13e3/src/Tribe/Main.php#L1432)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Main.php:1435](https://github.com/moderntribe/the-events-calendar/blob/9835f93e6c067ea78338cf99f5aa535c970c13e3/src/Tribe/Main.php#L1435)

> via jenkins/codesniffer-phpcs

**Pull request merged**